### PR TITLE
Version 1.1.0

### LINKED ISSUE #16: Add a default value for the dockerfiles list property
#### Is your feature request related to a problem? Please describe.**
If you use the sonar hadolint plugin without setting the `sonar.lang.patterns.dockerfile` property, nothing will be analysed as the property is needed. Projects might want to set a default value for this property to avoid setting it everytime.

#### Describe the solution you'd like
Add this property as a proper plugin property, and define a default value, so that it could be used or redefined inside the UI.

### LINKED ISSUE #17: Update the CI/CD process to use GitHub Actions
#### Is your feature request related to a problem? Please describe.**

See https://github.com/cnescatlab/sonar-cnes-report/issues/264 

#### Describe the solution you'd like

See https://github.com/cnescatlab/sonar-cnes-report/issues/264

### LINKED ISSUE #18: Update pom lib versions and SonarQube API
#### Is your feature request related to a problem? Please describe.**

SonarQube 8.9 is out so it's time refresh a bit our dependencies.

#### Describe the solution you'd like

Update the SonarQube API version we use, as well as all other dependencies.

### LINKED ISSUE #21: Update Hadolint & Shellcheck rules
#### Is your feature request related to a problem? Please describe.**

Since last release, Hadolint & ShellCheck may have evolved with new rules which are not included inside the plugin.

#### Describe the solution you'd like

Update the plugin rules to match the last available version of Hadolint & Shellcheck.

### LINKED ISSUE #26: Sonar scanner not detecting/scanning hadolint report for Dockerfiles
#### Describe what you need to know

I have installed the Sonarqube Hadolint plugin using the Marketplace in Sonarqube UI web interface.
I have a Bitbucket repository with many Dockerfiles scattered throughout in various directories. I run hadolint on each file and generate the hadolint-report.xml using `hadolint -f checkstyle < Dockerfile > hadolint-report.xml` and this file gets written in the same directory where the Dockerfile resides.
Then I run sonar-scanner using the Sonarqube Docker container in my repository with the following sonar-project.properties:
sonar.lang.patterns.dockerfile=Dockerfile
sonar.hadolint.reports.path=hadolint-report.xml

The sonar-scanner does the scans and uploads the data to my sonarqube instance but _nothing from hadolint appears_ nor do I see output during the sonar-scanner run that indicates it detects any hadolint-report.xml files or Dockerfiles.

#### User environment 
 - OS: Mac OSX and Linux
 - SonarQube version: 8.8.0
 - Java version: 11
 - Hadolint version: 2.6
 - Plugin version: 1.0.0

### LINKED ISSUE #27: Absolute report path for sonar.hadolint.reports.path
#### Describe what you need to know
I use this plugin in an Azure DevOps pipeline. Before I run SQ I import the hadolint.xml report as a pipeline artifact. 

Structure: 
Base directory (build.sourcesdirectory) is /home/vsts/work/1/s
```
/home/vsts/work/1/s
total 16
-rw-r--r-- 1 vsts docker   59 Aug 24 14:43 Dockerfile
-rw-r--r-- 1 vsts docker  985 Aug 24 14:43 README.md
-rw-r--r-- 1 vsts docker 2873 Aug 24 14:43 azure-pipelines.yml
-rw-r--r-- 1 vsts docker  379 Aug 24 14:43 sonar-project.properties
```
Artifact is by default imported in: /home/vsts/work/1
```
/home/vsts/work/1
total 20
drwxr-xr-x 2 vsts docker 4096 Aug 24 14:43 a
drwxr-xr-x 2 vsts docker 4096 Aug 24 14:43 b
drwxr-xr-x 2 vsts docker 4096 Aug 24 14:43 hadolint_results
drwxr-xr-x 3 vsts docker 4096 Aug 24 14:43 s
```
For `sonar.hadolint.reports.path` I specified the absolute path `/home/vsts/work/1/hadolint_results/hadolint.xml` However this doesn't work, it only accepts `../hadolint_results/hadolint.xml`

For other plugins report paths I specify the absolute path without an issue. Is this working as intended? 

#### User environment 
 - SonarQube version 8.9.0.43852
 - Hadolint version: v2.6.1

### PR DESCRIPTION
**Pull request by Sancretor**
## Proposed changes

Implements tasks & bugfixes for the 1.1.0 release.

## Types of changes

What types of changes does your code introduce to this plugin?

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Issues closed by changes

- [x] Close #16 
- [x] Close #17
- [x] Close #18 
- [x] Close #21
- [x] Close #26 
- [x] Close #27 

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/lequal/sonar-hadolint-plugin/blob/master/CONTRIBUTING.md) doc
- [x] I agree with the [CODE OF CONDUCT](https://github.com/lequal/sonar-hadolint-plugin/blob/master/CONTRIBUTING.md)
- [x] Lint and unit tests pass locally with my changes
- [x] SonarCloud and Travis CI tests pass with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <groupId>fr.cnes.sonarqube.plugins</groupId>
     <artifactId>sonar-hadolint-plugin</artifactId>
     <packaging>sonar-plugin</packaging>
-    <version>1.0.0</version>
+    <version>1.1.0-SNAPSHOT</version>
```
Comment (begarco) [inline pom.xml:8]: Is it intentional to you to append `-SNAPSHOT` ?

File: src/main/java/fr/cnes/sonar/plugins/hadolint/check/HadolintSensor.java [hunk 2]
```diff
@@ -196,22 +201,41 @@ static void saveIssue(final SensorContext context, final Map<String, InputFile>
...
+            includedFiles = scanner.getIncludedFiles();
+
+            if (includedFiles.isEmpty()) {
+                if (config.hasKey(reportPathPropertyKey)) {
+                    // try absolute path
+                    File file = new File(reportPath);
+                    if (!file.exists() || !file.isFile()) {
+                        LOGGER.warn("No report was found using pattern {}", reportPath);
+                    } else {
+                        result.add(file.getPath());
```
Comment (begarco) [inline src/main/java/fr/cnes/sonar/plugins/hadolint/check/HadolintSensor.java:229]: I do not understand how we can add reports here if includedFiles is empty ? Is it clear for you ?

**Review by begarco — COMMENTED** [review]
Two little questions.

File: pom.xml (hunk 1 above)
Comment (Sancretor) [reply pom.xml:8]: Well, that was put during the development phase, and I obviously forgot to remove this suffix.
I'll fix that.

File: src/main/java/fr/cnes/sonar/plugins/hadolint/check/HadolintSensor.java (hunk 2 above)
Comment (Sancretor) [reply src/main/java/fr/cnes/sonar/plugins/hadolint/check/HadolintSensor.java:229]: This is a special case inspired from a SonarQube official plugin : 

1. The includedFiles list contains files matching a given pattern (from the plugin property) and found with a dedicated method. 
2. If the property does not contain a pattern but a simple path (e.g. an absolute one), the matching method won't work. In this case, we try to match a unique file using the property as a path, not a pattern. If something matches, we add it to the list.
3. If even the simple path check fails, we don't add anything and in all cases, we notify this failure.

Is it understandable now ?

**Pull request merged**